inference-engine/src/transformations/src/transformations/smart_reshape/mimic_set_batch_size.cpp

```diff
@@ -2,35 +2,35 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include <ngraph/pass/constant_folding.hpp>
 #include <transformations/smart_reshape/mimic_set_batch_size.hpp>
 
 NGRAPH_RTTI_DEFINITION(ngraph::pass::MimicSetBatchSize, "MimicSetBatchSize", 0);
 
-ngraph::pass::MimicSetBatchSize::MimicSetBatchSize() {
-    auto reshape_label = ngraph::pattern::wrap_type<opset5::Reshape>({pattern::any_input(pattern::has_static_dim(0)),
-                                                                      ngraph::pattern::wrap_type<opset5::Constant>()},
-         [](const Output<Node> &output) { return output.get_partial_shape().rank().is_static() && output.get_partial_shape().rank().get_length() > 1; });
-
-    matcher_pass_callback callback = [=](pattern::Matcher &m) -> bool {
-        const auto & reshape = m.get_match_root();
-        auto pattern = std::dynamic_pointer_cast<opset5::Constant>(reshape->get_input_node_shared_ptr(1));
-        if (!pattern)
-            return false;
-
-        const auto & pattern_vector = pattern->cast_vector<int64_t>();
-        if (pattern_vector.empty() || pattern_vector[0] < 1)
-            return false;
-
-        // mimicking old setBatchSize style (copied):
-        // float diff = static_cast<float>(dims.at(0)) / static_cast<float>(originalBatchSize);
-        // dims.at(0) = static_cast<size_t>(std::ceil(size * diff));
-
-        const auto & old_input_batch = static_cast<float>(reshape->get_input_partial_shape(0)[0].get_length());
-        const auto & old_output_batch = static_cast<float>(pattern_vector[0]);
-
-        const auto & scale = old_output_batch / old_input_batch;
+bool ngraph::pass::MimicSetBatchSize::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    // extracting ratio of out to in 0-index dimension value from the folded function
+    auto specialized_function = ngraph::clone_function(*f);
+    ngraph::pass::ConstantFolding().run_on_function(specialized_function);
+    std::map<std::string, float> scale;
+    for (const auto & node : specialized_function->get_ops()) {
+        if (const auto & reshape = std::dynamic_pointer_cast<opset5::Reshape>(node)) {
+            const auto in_pshape = reshape->get_input_partial_shape(0), out_pshape = reshape->get_output_partial_shape(0);
+            if (in_pshape.rank().is_dynamic() || in_pshape.rank().get_length() <= 1 || in_pshape[0].is_dynamic() ||
+                out_pshape.rank().is_dynamic() || out_pshape.rank().get_length() <= 1 || out_pshape[0].is_dynamic())
+                continue;
+            const auto & pattern = std::dynamic_pointer_cast<opset5::Constant>(reshape->get_input_node_shared_ptr(1));
+            if (pattern && pattern->cast_vector<int64_t>()[0] > 0) {
+                scale[reshape->get_friendly_name()] = static_cast<float>(out_pshape[0].get_length()) / static_cast<float>(in_pshape[0].get_length());
```

Comment:
Can we store pointer or unique id in this map as a key instead of friendly_name?

Reply:
We can't as these are different functions